src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -123,6 +132,18 @@ enum TaskFailureReason {
     TASK_FAILURE_REASON_INVOCATION_ERROR = 3;
 }
 
+message ProgressUpdate {
+    float current = 1;
+    float total = 2;
```

Comment:
Why float?

Reply:
No particular reason; we needed something. Float gets us 23 bits of precision, plenty for most integer purposes.

Reply:
Floats are never "precise" afaik because they represent numbers using "steps". I undestand if we used float to represent fractions of something but we have current and total here so we can represent the full fraction with ints without loosing any precision due to floats. This is why I'd prefer ints or better uints here because we don't expect negative progress or total.

Reply:
Please confirm if libs like https://github.com/tqdm/tqdm use floats, then it's clear why

Reply:
Confirmed; TQDM takes either, progressbar2 only takes floats (but int should just cast over).